bindings/java/src/main/java/org/ucx/jucx/ucp/UcpParams.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2019. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+package org.ucx.jucx.ucp;
+
+import static org.ucx.jucx.ucs.UcsBits.UCS_BIT;
+
+/**
+ * Tuning parameters for UCP library.
+ *
+ * The structure defines the parameters that are used for
+ * UCP library tuning during UCP library {@link UcpContext} "initialization".
+ *
+ * <p>UCP library implementation uses the {@link UcpParams.UcpFeature} "features"
+ * parameter to optimize the library functionality that minimize memory
+ * footprint. For example, if the application does not require send/receive
+ * semantics UCP library may avoid allocation of expensive resources associated with
+ * send/receive queues.
+ */
+public class UcpParams {
+
+    @Override
+    public String toString() {
+        return "UcpParams{"
+            + "fieldMask=" + fieldMask
+            + ", features=" + features
+            + ", tagSenderMask=" + tagSenderMask
+            + ", mtWorkersShared=" + mtWorkersShared
+            + ", estimatedNumEps=" + estimatedNumEps
+            + '}';
+    }
+
+    /**
+     * UCP context parameters field mask.
+     *
+     * <p>The enumeration allows specifying which fields in {@link UcpParams} are
+     * present. It is used for the enablement of backward compatibility support.
+     */
+    private enum UcpParamField {
+        UCP_PARAM_FIELD_FEATURES(UCS_BIT(0L)),
+        // UCP_PARAM_FIELD_REQUEST_SIZE(UCS_BIT(1L)),
```

Comment:
minor: pls remove dead code

Reply:
I left this to keep semantic with native C, to show that there are such capabilities in UCP, so in the future maybe we could return to callbacks if would need somewhere.

Reply:
lets add this when it's needed